packages/@aws-cdk/aws-ec2/lib/volume.ts

```diff
@@ -129,6 +157,17 @@ export class BlockDeviceVolume {
    * @param options additional device options
    */
   public static ebs(volumeSize: number, options: EbsDeviceOptions = {}): BlockDeviceVolume {
+    // If KmsKeyId is specified, the encrypted state must be true.
+    if (options.kmsKeyId) {
+      if (options.encrypted === false) {
+        throw new Error('`encrypted` must be not false when providing `kmsKeyId`.');
+      } else {
+        // Encryption is implied if KMS key is specified.
+        (options.encrypted as unknown as boolean) = true;
```

Comment:
Please don't do this. Instead, reassign a new object that has the values you want to set:

```ts
options = {
   ...options,
  encrypted: true,
  kmsKeyId: options.kmsKey.keyArn,
};
```

Reply:
Fixed but I the tests fail now... -> https://github.com/aws/aws-cdk/pull/12842#issuecomment-801987291